Starting test execution, please wait...
Logging Vstest Diagnostics in file: C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\fine-code-coverage\diagnostics.log
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:01.87]     HospitalAppTests.ServicesTests.FeedbackServiceTests.Get_all_feedbacks [FAIL]
[xUnit.net 00:00:01.89]     HospitalAppTests.ServicesTests.FeedbackServiceTests.Get_by_id [FAIL]
  Failed HospitalAppTests.ServicesTests.FeedbackServiceTests.Get_all_feedbacks [4 ms]
  Error Message:
   System.NullReferenceException : Object reference not set to an instance of an object.
  Stack Trace:
     at HospitalApp.Services.FeedbackService.GetAll() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalApp\Services\FeedbackService.cs:line 25
   at HospitalAppTests.ServicesTests.FeedbackServiceTests.Get_all_feedbacks() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\FeedbackServiceTests.cs:line 25
  Failed HospitalAppTests.ServicesTests.FeedbackServiceTests.Get_by_id [2 ms]
  Error Message:
   System.NullReferenceException : Object reference not set to an instance of an object.
  Stack Trace:
     at HospitalApp.Services.FeedbackService.GetById(Int32 feedbackId) in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalApp\Services\FeedbackService.cs:line 32
   at HospitalAppTests.ServicesTests.FeedbackServiceTests.Get_by_id() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\FeedbackServiceTests.cs:line 33
[xUnit.net 00:00:11.37]     HospitalAppTests.ServicesTests.PatientServiceTests.Get_appointment_patient [FAIL]
  Failed HospitalAppTests.ServicesTests.PatientServiceTests.Get_appointment_patient [9 s]
  Error Message:
   System.ArgumentException : An item with the same key has already been added. Key: 22
  Stack Trace:
     at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryTable`1.Create(IUpdateEntry entry)
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryStore.ExecuteTransaction(IList`1 entries, IDiagnosticsLogger`1 updateLogger)
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(DbContext _, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.Storage.NonRetryingExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
[xUnit.net 00:00:11.43]     HospitalAppTests.ServicesTests.PatientServiceTests.Set_random_general_practitioner [FAIL]
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges()
   at HospitalAppTests.ServicesTests.PatientServiceTests.CreateDoctors(MyDbContext context) in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\PatientServiceTests.cs:line 198
   at HospitalAppTests.ServicesTests.PatientServiceTests.Get_appointment_patient() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\PatientServiceTests.cs:line 73
[xUnit.net 00:00:11.45]     HospitalAppTests.ServicesTests.UserServiceTests.Update_user [FAIL]
  Failed HospitalAppTests.ServicesTests.PatientServiceTests.Set_random_general_practitioner [62 ms]
  Error Message:
   Microsoft.EntityFrameworkCore.DbUpdateConcurrencyException : Attempted to update or delete an entity that does not exist in the store.
  Stack Trace:
     at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryTable`1.Delete(IUpdateEntry entry)
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryStore.ExecuteTransaction(IList`1 entries, IDiagnosticsLogger`1 updateLogger)
   at Microsoft.EntityFrameworkCore.InMemory.Storage.Internal.InMemoryDatabase.SaveChanges(IList`1 entries)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(IList`1 entriesToSave)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(DbContext _, Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.Storage.NonRetryingExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.ChangeTracking.Internal.StateManager.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges(Boolean acceptAllChangesOnSuccess)
   at Microsoft.EntityFrameworkCore.DbContext.SaveChanges()
   at HospitalAppTests.ServicesTests.PatientServiceTests.ClearDoctors(MyDbContext context) in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\PatientServiceTests.cs:line 222
   at HospitalAppTests.ServicesTests.PatientServiceTests.Set_random_general_practitioner() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\PatientServiceTests.cs:line 108
  Failed HospitalAppTests.ServicesTests.UserServiceTests.Update_user [36 ms]
  Error Message:
   System.NullReferenceException : Object reference not set to an instance of an object.
  Stack Trace:
     at HospitalAppTests.ServicesTests.UserServiceTests.Update_user() in C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\ServicesTests\UserServiceTests.cs:line 90
Failed!  - Failed:     5, Passed:    61, Skipped:     0, Total:    66, Duration: 45 s - HospitalAppTests.dll (netcoreapp3.1)

Calculating coverage result...
One or more errors occurred. (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.) (The process cannot access the file 'C:\Users\Super User\Documents\GitRepository\PSW\HospitalApp\HospitalAppTests\bin\Debug\netcoreapp3.1\HospitalApp.dll' because it is being used by another process.)
